src/Microsoft.ML.Transforms/Text/TextCatalog.cs

```diff
@@ -45,7 +45,7 @@ public static class TextCatalog
         /// </example>
         public static TextFeaturizingEstimator FeaturizeText(this TransformsCatalog.TextTransforms catalog,
             string outputColumnName,
-            IEnumerable<string> inputColumnNames,
```

Comment:
>  IEnumerable<string> inputColumnNames, [](start = 10, length = 39)

my vote, and the original issues is to have this be the last parameter of the method, and be a params, to optimize for both cases where the users has one or more columns and wants to provied options. 

Having the IEumerable,  or with the string[] will require building an Enumerable/Array to hold just one element, which is unnecessary since params exists. 

Why would the users have to do either:

` var customized_pipeline = ml.Transforms.Text.FeaturizeText(customizedColumnName, options, new List<string> { "SentimentText" }, 
var customized_pipeline = ml.Transforms.Text.FeaturizeText(customizedColumnName, options, new string[] { "SentimentText" },`

when they can just go:

 ` var customized_pipeline = ml.Transforms.Text.FeaturizeText(customizedColumnName, options,  "SentimentText"  `

I do think the signature should be:

 public static TextFeaturizingEstimator FeaturizeText(this TransformsCatalog.TextTransforms catalog,
            string outputColumnName,
            TextFeaturizingEstimator.Options options
            params string[] inputColumnNames,
            )

and we are doing that everywhere else...

Reply:
The assumption here is that output column is always single, but why can't it be multiple columns?

Reply:
it can be both, hence params is the cleanest 

---
In reply to: [261709072](https://github.com/dotnet/machinelearning/pull/2815#discussion_r261709072) [](ancestors = 261709072)